utils/build-script-impl

```diff
@@ -2604,7 +2608,7 @@ for host in "${ALL_HOSTS[@]}"; do
                 esac
 
                 ;;
-            foundation)
+            foundation|foundation_static)
```

Comment:
Should we build a static ICU as well?

Reply:
Yeah, that would be a good idea to do given the rest of the static linking, and the behaviour of glibc's `libdl`.  But, I think that is beyond the scope of this change.

Reply:
A static libICU is already built on linux as part of the `--libicu` option:
```
if [ $(true_false "${BUILD_SWIFT_STATIC_STDLIB}") == "TRUE" ]; then
         libicu_enable_static="--enable-static"
else
         libicu_enable_static=""
fi
```

Reply:
Oh, cool.  Um, why is that controlled by static stdlib though?  Lets get this working and then adjust the static build to use the static ICU.

Reply:
Since ICU is now bundled (presumably so that `"🏴󠁧󠁢󠁥󠁮󠁧󠁿".count == 1` instead of possibly 7) it's on us to build the static version of it as well. `-static-stdlib` changes the library search path to `lib/swift_static` instead of `lib/swift` so if there's no icu libs there it won't work.

Reply:
What should it be controlled by? Are we adding an extra `--static-` flag for each lib? maybe it would be better with a `--static-libs` flag instead to build them all together, `-static-stdlib` requires all of them anyway

Reply:
IOW it seems ICU is now considered part of the stdlib, so flags that control the linkage of the stdlib should cover ICU now too

Reply:
@kevints, this is about the build-script, not the the driver flag.  The driver flag remains `-static-stdlib`.  The icu libs are already present in the `swift_static` directory.

Reply:
Ok @compnerd I was confused then. This looks good to me